sycl/include/sycl/detail/cg.hpp

```diff
@@ -151,7 +160,7 @@ class CGExecKernel : public CG {
 public:
   /// Stores ND-range description.
   NDRDescT MNDRDesc;
-  std::unique_ptr<HostKernelBase> MHostKernel;
+  std::shared_ptr<HostKernelBase> MHostKernel;
```

Comment:
This change was made because we need to copy the command groups that we store in `node_impl`, wheras normally the runtime would move the command group when passing it to the scheduler. We can't guarantee that we wouldn't ever have to re-submit the command group (since we may have to rebuild the command buffers or build multiple command buffers from the same node) we can't really move it. 

We also wanted to avoid storing all possible required data for all command group types inside the node or otherwise duplicate a lot of existing functionality in the runtime.